src/osd/OSD.cc

```diff
@@ -3153,12 +3153,13 @@ PG *OSD::_create_lock_pg(
   return pg;
 }
 
```

Comment:
This isn't actually a race.  It's not obvious, but you need to hold the osd_lock (as well as the pg_map_lock) to modify the pg_map.  Holding either the pg_map_lock or the osd_lock is sufficient to ensure that it won't be modified.  The reason for this complexity is that the fast_dispatch path relies on being able to hold the pg_map_lock and not the osd_lock to read (but not modify) the pg_map.  That said, this is a perfectly good patch with the exception that you should return an rvalue PGRef.